### ISSUE TITLE
no-control-regex: Example of incorrect code on eslint.org doesn't trigger an error

### ISSUE THREAD

**Issue by nfriend** [buzinas/tslint-eslint-rules#60]
According to eslint.org's documentation on the `no-control-regex` rule (http://eslint.org/docs/rules/no-control-regex), the following should throw a linting error when `no-control-regex` is `true`:

```javascript
var pattern1 = /\\x1f/;
```

However, no error is thrown.  The second example in the documentation _does_ cause a linting error, as expected:
```javascript
var pattern2 = new RegExp("\x1f");
```

**Comment by nfriend** [comment]
@buzinas - I can take a look at this one unless you've already started on it.

**Comment by buzinas** [comment]
Didn't start yet. I took a look and I think the problem is that I'm only looking at the first token. So, probably if you loop through the tokens, it will be fixed (not sure if that's the problem, though). Thank you!

**Comment by buzinas** [comment]
I think it's an error in ESLint documentation.

It seems that there is no way to create a RegExp with control characters without the constructor. Look:

```
const x = new RegExp('\x1f');
const y = /\x1f/;
const z = /\\x1f/;
const w = /\\\x1f/;
const a = /\\\\x1f/;

console.log(x, y, z, w, a);

--- Logs
--- // /\x1f/ /\\x1f/ /\\\x1f/ /\\\\x1f/
```

**Comment by nfriend** [comment]
@buzinas I agree - I'll close this for now and wait on the resolution of the similar eslint issue I opened: https://github.com/eslint/eslint/issues/5737.
